include/circt/Dialect/FIRRTL/FIRRTLStatements.td

```diff
@@ -70,7 +70,7 @@ def StrictConnectOp : FIRRTLOp<"strictconnect", [SameTypeOperands, FConnectLike]
     "$dest `,` $src  attr-dict `:` qualified(type($dest))";
 }
 
-def RefDefineOp : FIRRTLOp<"ref.define", [SameTypeOperands, FConnectLike]> {
+def RefDefineOp : FIRRTLOp<"ref.define", [CompatibleRefTypes<"dest", "src">, FConnectLike]> {
```

Comment:
In the presence of the `RefCastOp`, this could be left `SameTypeOperands` ? Any particular reason, or just an IR design choice ?

Reply:
Is there any performance concerns for the recursive type verifier `isCompatibleRefType` ?